Jenkinsfile

```diff
@@ -10,20 +10,82 @@ pipeline {
     }
 
     tools {
-        maven 'apache-maven-latest' 
+        maven 'apache-maven-latest'
+        jdk 'openjdk-jdk11-latest'
+    }
+
+    environment {
+        PUBLISH_LOCATION = 'cbi/updates'
+    }
+
+    parameters {
+        choice(
+            name: 'BUILD_TYPE',
+            choices: ['nightly', 'milestone', 'release'],
+            description: '''
+                Choose the type of build.
+                Note that a release build will not promote the build, but rather will promote the most recent milestone build.
+            '''
+        )
+
+        booleanParam(
+           name: 'PROMOTE',
+            defaultValue: false,
+            description: 'Whether to promote the build to the download server.'
+        )
+
+        booleanParam(
+            name: 'ARCHIVE',
+            defaultValue: false,
+            description: 'Whether to archive the workspace.'
+        )
     }
 
     stages {
+        stage('Display Parameters') {
+            steps {
+                echo "BUILD_TYPE=${params.BUILD_TYPE}"
+                echo "PROMOTE=${params.PROMOTE}"
+                echo "ARCHIVE=${params.ARCHIVE}"
+                echo "GIT_COMMIT=${env.GIT_COMMIT}"
+                script {
+                    env.PROMOTE = params.PROMOTE
+                    env.BUILD_TYPE = params.BUILD_TYPE
+                }
+             }
+        }
+    
         stage('Build') {
             steps {
                 dir('scm') {
-                    wrap([$class: 'Xvnc', takeScreenshot: false, useXauthority: true]) {
-                        sh 'mvn -B clean install -Ptarget-default'
+                    sshagent(['projects-storage.eclipse.org-bot-ssh']) {
```

Comment:
This is an abuse of `projects-storage`. No build should run there. We allow some projects to do minor stuff over there so that they did not have to rewrite their whole publishing workflow, but we certainly don't want to promote this.

Reply:
How is this an abuse?

The whole point of this is to be able to do promotion of the update sites easily from within a build to produce this:

https://download.eclipse.org/cbi/updates/tpd/

This involves launching from within Maven an Eclipse application to do that publishing, which uses rsync  within a Java application. 

What aspect of this is abusive and what is the negative impact of that abuse?

Reply:
I'm afraid I've read it too fast, I thought the `mvn` step was being executed on `projects-storage`. This is all good. Thanks for elaborating!

Reply:
I see!  No that would be horrible indeed! I"m relieved because I do similar things for JustJ for the CBI aggregator...